packages/components/src/List/ListComposition/SortBy/SortBy.component.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Navbar, NavDropdown, Nav, NavItem, MenuItem } from 'react-bootstrap';
+import uuid from 'uuid';
+import isUndefined from 'lodash/isUndefined';
+
+import { useListContext } from '../context';
+
+function SortBy(props) {
+	const { id, options, selected, isDescending: isDescendingProp, onChange, onOrderChange } = props;
+	const { sortParams, setSortParams, t } = useListContext();
+
+	// Sort by
+	let selectedOption;
+	if (selected) {
+		selectedOption = selected;
+	} else if (sortParams.sortBy) {
+		selectedOption = sortParams.sortBy;
+	} else {
+		selectedOption = options[0].key;
+
+		useEffect(() => {
+			setSortParams({ ...sortParams, sortBy: selectedOption });
+		});
+	}
+
+	const onSelect = onChange
```

Comment:
I may miss a point here.
You can pass a props onChange to apply the filter you want, but in the list manager we are always calling useMemo(() => sortCollection. 
I'm not sure what the workflow will be when I will made a custom sort...

Reply:
Correct, I guess it would mess with the sorting memoization since `collection` would change.
But I'm not sure on how we could prevent this from happening (expect from controlling `ListManager`'s behavior on sorting/filtering via its props), since `ListManager` and `SortBy` are not supposed to communicate upwards (`SortBy` > `ListManager`)